sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala

```diff
@@ -240,6 +280,10 @@ case class OptimizeSkewedJoin(conf: SQLConf) extends Rule[SparkPlan] {
       rightStats: MapOutputStatistics,
       nonSkewPartitionIndices: Seq[Int]): Seq[ShufflePartitionSpec] = {
     assert(nonSkewPartitionIndices.nonEmpty)
+    if (!conf.getConf(SQLConf.REDUCE_POST_SHUFFLE_PARTITIONS_ENABLED)) {
```

Comment:
not related to this PR but a small fix: we shouldn't coalesce partitions if the config is off.